WalletWasabi.Fluent/Models/WebcamQrReader.cs

```diff
@@ -0,0 +1,164 @@
+using Avalonia.Media.Imaging;
+using System;
+using System.Threading.Tasks;
+using OpenCvSharp;
+using Avalonia;
+using Avalonia.Media;
+using System.Runtime.InteropServices;
+using WalletWasabi.Logging;
+using WalletWasabi.Userfacing;
+using NBitcoin;
+using Nito.AsyncEx;
+using Avalonia.Platform;
+
+namespace WalletWasabi.Fluent.Models
+{
+	public class WebcamQrReader
+	{
+		private const byte DefaultCameraId = 0;
+
+		public WebcamQrReader(Network network)
+		{
+			ScanningTaskLock = new();
+			Network = network;
+		}
+
+		public event EventHandler<WriteableBitmap>? NewImageArrived;
+
+		public event EventHandler<string>? CorrectAddressFound;
+
+		public event EventHandler<string>? InvalidAddressFound;
+
+		public event EventHandler<Exception>? ErrorOccured;
+
+		private AsyncLock ScanningTaskLock { get; set; }
+		public bool RequestEnd { get; set; }
+		public Network Network { get; }
+		public Task? ScanningTask { get; set; }
+		public bool IsRunning => ScanningTask is not null;
+
+		public async Task StartScanningAsync()
+		{
+			using (await ScanningTaskLock.LockAsync().ConfigureAwait(false))
+			{
+				if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+				{
+					ErrorOccured?.Invoke(this, new NotImplementedException("This operating system is not supported."));
+				}
```

Comment:
Is this worth it? Isn't it easier to just hide the QR reading feature in UI or say in UI that the feature is not available.

Reply:
On the UI it will just be a notification, and the exception itself should be a reminder that in the future this needs to be implemented.
On the other hand,  
>  Isn't it easier to just hide the QR reading feature in UI

This might be something to consider

Reply:
#6171